### PR TITLE
Update copy function- Use fileHandleId

### PR DESCRIPTION
**Pull request by thomasyu888**

### CODE REVIEW

File: synapseclient/__main__.py [hunk 1]
```diff
@@ -219,12 +220,22 @@ def copy(args,syn):
     for i in search:
         if i['file.name'] == ent.name:
             raise ValueError('Filename exists in directory you would like to copy to, either rename or check if file has already been copied!')
-    new_ent = synapseclient.File(ent['path'],parent=args.parentid)
+    #CHECK: If the user created the file, copy the file by using fileHandleId else hard copy
+    if profile == ent.properties.createdBy:
+        tempFile = utils.make_bogus_data_file()
```
Comment (cbare) [inline synapseclient/__main__.py]: It shouldn't be necessary to make and throw away a dummy file here. Try something like:

```python
new_ent = File(name=ent.name, description=ent.description, parentId=args.parentId)
new_end.dataFileHandleId = ent.dataFileHandleId
```

File: synapseclient/__main__.py [hunk 2]
```diff
@@ -219,12 +220,22 @@ def copy(args,syn):
     for i in search:
         if i['file.name'] == ent.name:
             raise ValueError('Filename exists in directory you would like to copy to, either rename or check if file has already been copied!')
-    new_ent = synapseclient.File(ent['path'],parent=args.parentid)
+    #CHECK: If the user created the file, copy the file by using fileHandleId else hard copy
+    if profile == ent.properties.createdBy:
```
Comment (cbare) [inline synapseclient/__main__.py]: Could it ever happen that a File was created by the current user, but it's fileHandle wasn't?

Comment (thomasyu888) [inline synapseclient/__main__.py]: When you try to copy a file that you don't own, an error will the thrown if you try to set fileHandleId.

Comment (larssono) [inline synapseclient/__main__.py]: Use: syn.restGET('/entity/%s/filehandles' %f.id) to get the filehandles and check the createdBy

Comment (larssono) [inline synapseclient/__main__.py]: OR use GET /fileHandle/{handleId} and go over to download/upload of the file if you get a 403 error back

**Pull request merged**